share/chruby/chruby.sh

```diff
@@ -13,7 +13,7 @@ function chruby_reset()
 	PATH=":$PATH:"; PATH="${PATH//:$RUBY_ROOT\/bin:/:}"
 	[[ -n "$GEM_ROOT" ]] && PATH="${PATH//:$GEM_ROOT\/bin:/:}"
 
-	if (( UID != 0 )); then
+	if (( UID != 0 )) && [ "$CHRUBY_SET_GEM_HOME" != "false" ]; then
```

Comment:
Other tests expressions use double brackets, which I think would be best to stick with for consistent compatibility and style.

My first impulse was to prefer `0` over `"false"` for falsey, but I don't have a strong defense.

Reply:
OK, I'll switch to double brackets.

I think `false` is a bit more readable as in:
```bash
# In e.g., ~/.profile
export CHRUBY_SET_GEM_HOME=false

# Looks more cryptic to me:
export CHRUBY_SET_GEM_HOME=0
```

Reply:
Done, please re-review.